distribution/integration/test-common/integration-test-utils/src/main/java/org/wso2/carbon/integration/common/utils/CarbonCommandToolsUtil.java

```diff
@@ -0,0 +1,485 @@
+            });
+            return process;
+
+        } catch (IOException ex) {
+            log.error("Error when reading the InputStream when running shell script ", ex);
+            throw new CarbonToolsIntegrationTestException("Error when reading the InputStream when " +
+                                                          "running shell script ", ex);
+        }
+    }
+
+    /**
+     * This method to find multiple strings in same line in log
+     *
+     * @param stringArrayToFind - String array to find all the elements
+     * @return boolean - true if found all the strings , false if not
+     */
+    public static boolean findMultipleStringsInLog(String[] stringArrayToFind) {
+        boolean expectedStringFound = false;
+
+        long startTime = System.currentTimeMillis();
+        while (!expectedStringFound && (System.currentTimeMillis() - startTime) < CarbonIntegrationConstants.DEFAULT_WAIT_MS) {
+            String message = serverLogStreamHandler.getOutput();
+            for (String stringToFind : stringArrayToFind) {
+                if (message.contains(stringToFind)) {
+                    expectedStringFound = true; // Continue until find all the strings in this line
+                } else {
+                    expectedStringFound = false;
+                    break; // break the loop and search the stringToFind in new line
+                }
+            }
+            try {
+                Thread.sleep(500); // wait for 0.5 second to check the log again.
+            } catch (InterruptedException e) {
+                log.warn("Exception while waiting for log message");
+            }
+        }
+        return expectedStringFound;
+    }
+
+    /**
+     * This method to check whether server is up or not
+     * This method wait for some time to check login status by checking the port and login
+     * This will throw an exception if port is not open or couldn't login
+     *
+     * @param automationContext - AutomationContext
+     * @return true: If server is up else false
+     * @throws CarbonToolsIntegrationTestException - Error while waiting for login
+     */
+    public static boolean isServerStartedUp(AutomationContext automationContext, int portOffset)
+            throws CarbonToolsIntegrationTestException {
+        try {
+            //Waiting util a port is open, If couldn't open within given time this will throw an Exception
+            ClientConnectionUtil.waitForPort(
+                    Integer.parseInt(FrameworkConstants.SERVER_DEFAULT_HTTPS_PORT) + portOffset,
+                    CarbonIntegrationConstants.DEFAULT_WAIT_MS, false,
+                    automationContext.getInstance().getHosts().get("default"));
+
+            //Waiting util login to the the server this will throw LoginAuthenticationExceptionException if fails
+            ClientConnectionUtil.waitForLogin(automationContext);
+            log.info("Server started successfully.");
+            return true;
```

Comment:
waitForLogin throws Exception, at the moment i don't have control on it i discussed about this with the team, we will fix it in the next test framework release to send custom or specific exception properly.